c/pthread-C-DAC/pthread-producer-consumer.c

```diff
@@ -0,0 +1,227 @@
+/*****************************************************************************
+                        C-DAC Tech Workshop : hyPACK-2013                             
+ October 15-18,2013
+ 
+  Example     : pthread-producer-consumer.c
+  
+  Objective   : To illustrate producer-consumer problem using pthreads.
+ 
+  Input       : Nothing.
+ 
+  Output      : Sequence of Produced and Consumed items.
+                                                                                                                         
+  Created     : MAY-2013 
+  E-mail      : hpcfte@cdac.in          
+                                
+****************************************************************************/
+ 
+#include <pthread.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <stdlib.h>
+ 
+/*defining QUEUE sise */
+#define QUEUESIZE 10
+#define LOOP 20
+ 
+/*global variable declaration */
+int ret_count;
+/*Thread callback function prototype*/
+void *producer (void *args);
+void *consumer (void *args);
+ 
+typedef struct {
+         
+            int buf[QUEUESIZE];
+            long head, tail;
+            int full, empty;
+            pthread_mutex_t *mut;
+            pthread_cond_t *notFull, *notEmpty;
+           } queue;
+ 
+ 
+ 
+queue *queueInit (void);
+void queueDelete (queue *q);
+void queueAdd (queue *q, int in);
+void queueDel (queue *q, int *out);
+ 
+/*Main function */
+int main ()
+{
+    queue *fifo;
+    pthread_t pro, con;
+    /*QUEUE initialization */
+    fifo = queueInit ();
+    if (fifo == NULL)
+        {
+        fprintf (stderr, "main: Queue Init failed.\n");
+        exit (1);
+    }
+    /*producer thread creation */
+    ret_count=pthread_create (&pro, NULL, producer, fifo);
+        if(ret_count)
+        {
+        printf("\n ERROR : Return code from pthread_create() is %d ",ret_count);
```

Comment:
hm, maybe the intent of this line and the following exit() was to always guarantee the success of pthread_create. I do not know, if the exit here would not even be correct. The main goal/property of this source file is related to the producer-consumer-part, and not to creating a thread.

@dbeyer , @PhilippWendler : Any suggestions?